docs/customization.md

```diff
@@ -186,6 +186,27 @@ def get_resolver_provider(ctx, req, include_sdists, include_wheels, sdist_server
     ...
 ```
 
+The `GenericProvider` is a convenient base class, or can be instantiated
+directly if given a `version_source` callable that returns an iterator of
+version values.
+
+```python
+from fromager import resolver
+
+VERSION_MAP = {'1.0': 'first-release', '2.0': 'second-release'}
+
+def _version_source(
+        identifier: str,
+        requirements: resolver.RequirementsMap,
+        incompatibilities: resolver.CandidatesMap,
+    ) -> typing.Iterable[Version]:
+    return VERSION_MAP.keys()
```

Comment:
In this example `VERSION_MAP` keys are `str`, not `Version` objects.

Reply:
The values can be str or Version objects. I'll clarify that.